lib/html.php

```diff
@@ -12,6 +12,41 @@
  * @link    https://github.com/rss-bridge/rss-bridge
  */
 
+function render(string $template, array $context = []): string
+{
+    $context['page'] = render_template($template, $context);
+    return render_template('base.html.php', $context);
+}
+
+function render_template(string $template, array $context = []): string
+{
+    extract($context);
+    ob_start();
+    try {
+        require __DIR__ . '/../templates/' . $template;
+    } catch (\Throwable $e) {
+        ob_end_clean();
+        throw $e;
+    }
+    return ob_get_clean();
+}
+
+/**
+ * Escape for html context
+ */
+function e(string $s): string
+{
+    return htmlspecialchars($s, ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8', false);
```

Comment:
Why `double_encode=false`? Either `$s` is a plain text and then entities should be escaped (e.g. so that `&amp; is escape sequence for ampersand` is displayed correctly), or it is a HTML string, in which case it will be inconsistently mangled as well.

Reply:
`double_encode=false` makes sure that `e(e('&'))` still is `&amp;` and not `&amp;amp;`. I thought it was nice but yeah might be confusing and buggy.